### PR TITLE
Keeper tests

### PR DESCRIPTION
**Pull request by musnit**

### CODE REVIEW

**Comment by musnit** [comment]
Note: Test fails intentionally on this branch as code not yet implemented. Later pull request has implementation with all passing tests.

File: x/ethbridge/types/ethbridgeclaim.go [hunk 1]
```diff
@@ -60,11 +60,12 @@ func CreateEthClaimFromOracleClaim(nonce int, ethereumSender string, validator s
 		return EthBridgeClaim{}, sdk.ErrInternal(fmt.Sprintf("failed to parse claim: %s", errRes))
 	}
 
+	valAccAddress := sdk.AccAddress(validator)
 	return NewEthBridgeClaim(
 		nonce,
 		ethereumSender,
 		oracleClaim.CosmosReceiver,
-		validator,
+		valAccAddress,
```
Comment (denalimarsh) [inline x/ethbridge/types/ethbridgeclaim.go:68]: +1 for increased explicitness

File: x/oracle/keeper/keeper.go [hunk 2]
```diff
@@ -92,40 +93,34 @@ func (k Keeper) ProcessClaim(ctx sdk.Context, id string, validator sdk.AccAddres
...
-func (k Keeper) getPowerThreshold() int {
-	minimumPower := float64(getTotalPower()) * k.consensusNeeded
-	return int(math.Ceil(minimumPower))
-
-}
-
-func getTotalPower() int {
-	//TODO: Get from Tendermint/last block/staking module?
-	return 10
+func (k Keeper) processCompletion(ctx sdk.Context, prophecy types.Prophecy) types.Prophecy {
```
Comment (denalimarsh) [inline x/oracle/keeper/keeper.go:111]: Thoughts on extracting some of these logic operations out to a library for clarity?

File: x/oracle/keeper/test_common.go [hunk 3]
```diff
@@ -105,32 +121,29 @@ func createTestAddrs(numAddrs int) []sdk.AccAddress {
...
+	var publicKeys []crypto.PubKey
+	var buffer bytes.Buffer
 
-	bechres, err := sdk.AccAddressFromBech32(bech)
-	if err != nil {
-		panic(err)
-	}
-	if bytes.Compare(bechres, res) != 0 {
-		panic("Bech decode and hex decode don't match")
+	//start at 10 to avoid changing 1 to 01, 2 to 02, etc
```
Comment (denalimarsh) [inline x/oracle/keeper/test_common.go:138]: This should be fine since it's a testing file.

**Review by denalimarsh — COMMENTED** [review]
Looks good. Review comments related to introducing a dedicated library to handle some of the mathematical calculations for increased clarity.

File: x/oracle/keeper/test_common.go [hunk 4]
```diff
@@ -52,16 +68,76 @@ func CreateTestKeepers(t *testing.T, isCheckTx bool, consensusNeeded float64, in
...
+		panic(err)
+	}
+	if bytes.Compare(bechres, res) != 0 {
+		panic("Bech decode and hex decode don't match")
+	}
+
+	return res
 }
 
 // MakeTestCodec creates a codec used only for testing
```
Comment (denalimarsh) [inline x/oracle/keeper/test_common.go:149]: Testing Codec was primarily introduced for simplification purposes?

File: x/oracle/keeper/keeper.go (hunk 2 above)
Comment (musnit) [reply x/oracle/keeper/keeper.go:111]: I think it's fine for now as it's just this 1 function

File: x/oracle/keeper/test_common.go (hunk 4 above)
Comment (musnit) [reply x/oracle/keeper/test_common.go:149]: Yeh, follows same way done in other cosmos modules

**Pull request merged**